homeassistant/components/madvr/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for the integration."""
+
+import asyncio
+import logging
+from typing import Any
+
+import aiohttp
+from madvr.errors import CannotConnect
+from madvr.madvr import HeartBeatError, Madvr
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PORT
+
+from .const import DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_HOST,
+        ): str,
+        vol.Required(
+            CONF_PORT,
+            default=DEFAULT_PORT,
+        ): int,
+    }
+)
+
+
+class MadVRConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the integration."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input[CONF_PORT]
+
+            # prevent multiple instances with same host
+            await self.async_set_unique_id(host)
+            self._abort_if_unique_id_configured()
+
+            try:
+                # get the mac address from device
+                mac = await self._test_connection(host, port)
```

Comment:
using self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST], CONF_PORT: user_input[CONF_PORT]})

because mac may or may not be available (99% probably but not 100) also there would be no reason to configure two of the same hosts anyway

Reply:
There is no reason to configure two of the same, but if people are able to, they can create a mess so we rather try our best to avoid such situatons

Reply:
okay makes sense I will grab the mac first, create a unique ID based on that, and abort. Also will use that for enttity unique IDs as you mentioned below

Reply:
Awesome!